samples/vmc/networks_nsxt/l3_vpn_crud.py

```diff
@@ -0,0 +1,266 @@
+                ike_encryption_algorithms=[L3Vpn.IKE_ENCRYPTION_ALGORITHMS_128],
+                # Value should be per the user setup config
+                ike_version=L3Vpn.IKE_VERSION_V1,  # Value should be per the user setup config
+                l3vpn_session=l3vpn_session,
+                local_address=local_end_point_ip,
+                passphrases=[self.args.passphrase],
+                remote_public_address=self.args.remote_endpoint_public_ip,
+                tunnel_digest_algorithms=[L3Vpn.TUNNEL_DIGEST_ALGORITHMS_SHA1],
+                # Value should be per the user setup config
+                tunnel_encryption_algorithms=[L3Vpn.TUNNEL_ENCRYPTION_ALGORITHMS_128]
+                # Value should be per the user setup config
+            )
+            self.nsx_client.infra.tier_0s.locale_services.L3vpns.patch("vmc", "default",
+                                                                       l3vpn_id=vpn_id, l3_vpn=self.l3VPN)
+        except Exception as ex:
+            print(ex)
+            self.log_error(ex)
+
+    def create_route_based_l3_vpn(self, vpn_id):
+        print(' Create route based L3VPN '.center(70, '='))
+        try:
+            context = self.get_l3_vpn_context()
+            local_end_point_ip = context.available_local_addresses[0].address_value
+            print "local_end_point_ip={}".format(local_end_point_ip)
+            tunnel_subnet = TunnelSubnet(ip_addresses=["169.254.2.1"],  # Value should be per the user setup config
+                                         prefix_length=24)  # Value should be per the user setup config
+            bgpconfig1 = BgpNeighborConfig(links=None,
+                                           description="bgp neighbor config",
+                                           display_name="bgp_neighbor_config_1",
+                                           id="bgp_neighbor_config_1",
+                                           neighbor_address="169.254.2.2",  # Value should be per the user setup config
+                                           remote_as_num=str(65002))  # Value should be per the user setup config
+            self.nsx_client.infra.tier_0s.locale_services.bgp.Neighbors.patch(tier0_id="vmc",
+                                                                              locale_service_id="default",
+                                                                              neighbor_id="rb_neighbor_1",
+                                                                              bgp_neighbor_config=bgpconfig1)
+
+            neighbor_list = self.nsx_client.infra.tier_0s.locale_services.bgp.Neighbors.list(tier0_id="vmc",
+                                                                                             locale_service_id="default"
+                                                                                             )
+            print "List of neighbors={}".format(neighbor_list)
```

Comment:
This only works in python2. Could you please test the samples in python3? It'd be great if you could post result in both python2 and python3.

Reply:
Missed adding the changes after testing with python3 in last commit.